src/target/xtensa/xtensa_fileio.c

```diff
@@ -0,0 +1,201 @@
+/***************************************************************************
```

Comment:
Can you leave semihosting feature to the separate patch? 

I have a patch https://review.openocd.org/c/openocd/+/6889. Please check this.

Reply:
Separated.  I will follow up on semihosting vs. fileio later